lib/structuraid_core/design_codes/aci31819/rc/footings/maximum_rebar_spacing.rb

```diff
@@ -0,0 +1,66 @@
+module StructuraidCore
+  module DesignCodes
+    module Aci31819
+      module Rc
+        module Footings
+          class MaximumRebarSpacing
+            include DesignCodes::Utils::CodeRequirement
+            use_schema DesignCodes::Schemas::Rc::Footings::MaximumRebarSpacingSchema
+
+            MIN_HEIGHT_MAPPINGS = {
+              over_soil: 150,
+              over_piles: 300
+            }.freeze
+
+            CODE_REFERENCE = 'NSR-10 c.10.5.4 and C.10.6.4'.freeze
+
+            def call
+              unless MIN_HEIGHT_MAPPINGS.keys.include?(support_type)
+                raise Errors::DesignCodes::UnrecognizedValueError.new(:support_type, support_type)
+              end
```

Comment:
we can use the `enum` we built on the schema definition, it will automatically raise this error y support_type is not valid

Reply:
yeah, done!